integrations/Iterable/utils.js

```diff
@@ -0,0 +1,40 @@
+import { getDataFromSource } from "../../utils/utils";
+
+
+const ITEMS_MAPPING = [
+  { src: "product_id", dest: "id" },
+  { src: "sku", dest: "sku" },
+  { src: "name", dest: "name" },
+  { src: "price", dest: "price" },
+  { src: "quantity", dest: "quantity" },
+  { src: "image_url", dest: "imageUrl" },
+  { src: "url", dest: "url" }
+]
+
+function getMappingObject (properties, mappings) {
+    let itemsObject = {};
+    mappings.forEach((mapping) => {
+        itemsObject = {
+        ...getDataFromSource(mapping.src, mapping.dest, properties),
+        ...itemsObject,
+      };
+    });
+    return itemsObject;
+}
+
+function formPurchaseEventPayload(message) {
+    let purchaseEventPayload = {};
+    const { products } = message.properties;
```

Comment:
We are considering events where there is products array what about events where there is product info on properties root